### ISSUE TITLE
Posts are not generated in production mode

### ISSUE THREAD

**Issue by anagstef** [GatsbyCentral/gatsby-starter-wordpress#35]
Hello everyone!

Everything works fine in dev environment (`gatsby develop`) but when I build for production there is an issue. When I run `gatsby build` and then `gatsby serve` every post page is not generated. You can't open any post, you get a 404.

In fact at build time it prints this warning:

```
warning The GraphQL query in the non-page component "/Users/anagstef/repos/test-gatsby-build/src/templates/post.js" will not be run.
Exported queries are only executed for Page components. Instead of an exported
query, either co-locate a GraphQL fragment and compose that fragment into the
query (or other fragment) of the top-level page that renders this component, or
use a <StaticQuery> in this component. For more info on fragments and
composition, see http://graphql.org/learn/queries/#fragments and for more
information on <StaticQuery>, see https://next.gatsbyjs.org/docs/static-query
```

Happy to help with this, if I can do it (I have no experience with GraphQL).

**Comment by anagstef** [comment]
@chmac Just tried that and nothing changed :(

**Comment by chmac** [comment]
@anagstef Have you changed the WordPress site to your own URL? Are you using authentication?

I've just retested this repo, without making any changes, and `yarn build && yarn serve` gives me the expected result. I can see multiple posts, no error messages.

Is your repo public? Can you share the URL to see your code? I'm at a loss as to what to suggest otherwise I'm afraid.

**Comment by chmac** [comment]
@anagstef Ignore my last message, on a fresh install, I see the same bug locally. Investigating now.

**Comment by anagstef** [comment]
@chmac Yes, I was talking about a fresh install from this repo, no changes at all. I think it worked a few days ago. Maybe some dependency update broke something?

**Comment by chmac** [comment]
@anagstef Arg, bug was in 638379a42d0829f1eba71433ada5fadd10206822. Added filtering but forgot to add the required field for the filter. The fix is one line in ef8ccc14185c34a09383110dc3f2e572c6e0ca46.

Hopefully that change works. I'll close this issue now, but if it doesn't, please reopen and I'll look again. Thanks for taking the time to report the issue, I appreciate it.
